src/bootstrap/src/core/builder/cargo.rs

```diff
@@ -654,6 +654,7 @@ impl Builder<'_> {
                 // supported by the target.
                 if target != compiler.host && cmd_kind != Kind::Check {
                     let error = command(self.rustc(compiler))
+                        .allow_failure()
```

Comment:
you should also update the `not_supported` check in this case to be `command.success() && !line.contains("unsupported crate type")`. otherwise a command that fails to run will be treated as success.

Reply:
Before https://github.com/rust-lang/rust/pull/130899, this was set unconditionally, I wanted failed builds to have the previous behaviour as before.

Reply:
This can't be a concern anymore with https://github.com/rust-lang/rust/pull/136309/commits/139d6ba054a1a4cc5fe64981ad46fd5547bd4916.